### PR TITLE
Feature/thunk

### LINKED ISSUE #36: Create thunks for onLoad on student dash
- [x] Create thunk for getting the lesson
- [x] Create thunk for getting the students

### LINKED ISSUE #47: Write a thunk GET request, to retrieve teacher id, first and last name
- [x] Write a thunk GET request, to retrieve teacher id, students, lessons on login (or onLoad?) for teacher

### LINKED ISSUE #50: Write a thunk GET request to get the lesson for the student
- [x] Write a thunk GET request to get the lesson for the student

### PR DESCRIPTION
**Pull request by Billwilke42**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- No thunks were written and no setTeacher action.

closes #47 
closes #50 
closes #36
## What is the new behavior?

- Thunks were written and setTeacher action 
## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
<<<<<<< HEAD

=======

>>>>>>> 49ea6a2e4d81cd8aaf5ad430b688adb405c87982

### CODE REVIEW

File: src/StudentDashboard/StudentDashboard.js [hunk 1]
```diff
@@ -16,12 +16,12 @@ function StudentDashboard(props) {
  return (
     <section className='student-body'>
       {props.lessonOver && <MoodForm />}
-      {!props.studentName && <StudentForm />}
-      {(props.studentName && !props.lessonOver) && 
+      {!props.student && <StudentForm />}
```
Comment (KyleWong2510) [inline src/StudentDashboard/StudentDashboard.js:19]: weird one for sure

**Review by KyleWong2510 — APPROVED** [review]
Great job with all your tests!  Especially the thunks.

File: src/StudentDashboard/QuestionArea.js [hunk 2]
```diff
@@ -20,7 +20,9 @@ function QuestionArea(props) {
   shuffle(answersCopy)
   
   const answerClick = (e) => {
-    if(e.target.value) {
+    debugger
+    console.log(e.target.value === "true")
```
Comment (Melizzo) [inline src/StudentDashboard/QuestionArea.js:24]: May need to remove this debugger

File: src/thunks/__tests__/getTeacher.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      "last_name": "test teacher last"
+    }
+    mockUrl = `https://gumberoo-backend.herokuapp.com/api/v1/teachers/${mockTeacherId}`
+    mockDispatch = jest.fn()
+    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
+      ok: true,
+      json: () => Promise.resolve({
+        data: mockTeacher
+      })
+    }))
```
Comment (Melizzo) [inline src/thunks/__tests__/getTeacher.js:25]: These are always so much fun - but nice work Bill!!

File: src/thunks/getLesson.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { isLoading, hasErrored, setLesson } from '../actions'
+
+export const getLesson = (teacherId, lessonId) => {
+  const url =` https://gumberoo-backend.herokuapp.com/api/v1/teachers/${teacherId}/${lessonId}`
+
+  return async (dispatch) => {
+    try {
+      dispatch(isLoading(true))
```
Comment (Melizzo) [inline src/thunks/getLesson.js:8]: Nice work on the thunks here

**Review by Melizzo — APPROVED** [review]

**Pull request merged**